pandas/core/internals/blocks.py

```diff
@@ -3038,10 +3040,12 @@ def shift(self, periods, axis=0):
 
         new_values = self.values.asi8.take(indexer)
 
+        if fill_value is None:
```

Comment:
use ``if isna(fill_value)``, I think you have another occurrence of this as well